pkg/plugin/plugin.go

```diff
@@ -631,9 +541,9 @@ func CmdDel(args *skel.CmdArgs) error {
 
 	// Delete can be called multiple times, so don't return an error if the
```

Comment:
IIUIC this is not the case anymore due to the early return if the cache file is missing. So in theory, we could drop the special handling here. On the other hand, then we rely on the cache logic as some sort of side effect. I'm not sure which route to take, just bringing this up in case you have a preference - I don't mind neither.

Reply:
yes @phoracek , now changed it to rely on cache to restore original state as much as possible in case of error.
if this looks ok for you, please do `/hold cancel` :)